doc/media-settings/Media-based-Port-settings.md

```diff
@@ -0,0 +1,313 @@
+Media based port settings in SONIC
+
+# 1 Document History
+
+| Version | Date       | Author                 | Description                                      |
+|---------|------------|------------------------|--------------------------------------------------|
+| v.01    | 01/30/2019 |Sudharsan               | Initial version from Dell                        |
+
+Overview:
+=========
+
+              Networking devices such as switches or routers use different kind of optics as well as DAC cables to connect with peer devices such as servers, storage or other networking devices. There are different types of networking cables available and based on the cable type the networking ASIC might require additional settings for a cable to work properly on a port. As an example, a 40G-QSFP-CR4 2M DAC might require a different pre-emphasis setting on a device based on Broadcom ASIC than 40G-QSFP-CR4 1M DAC on a port. If the pre-emphasis is not set properly the traffic flowing through the port might encounter CRC errors and even in some cases the port might not even become operationally up.
+
+              To solve the above problem, few SAI attributes required for serdes programming are listed in section below. These SAI attributes will be programmed during media insert event.
+
+Media settings file:
+====================
+
+              Each vendor need to define the media settings file under device/<vendor-name>/ <ONIE_PLATFORM_STRING>/ <HARDWARE_SKU>/media_settings.json. This file contains the list of optics and DAC media-based settings for each port in the device. Each media setting for a media type comprises of key-value pairs per lane that is expected to be programmed when an optics or DAC is used. Examples of key value pairs include pre-emphasis, idriver and ipredriver.
```

Comment:
can we have a way for some vendors to bypass this mechanism?  They may have other ways to tune the SFP.

Reply:
Hi @keboliu . Yes. If the vendor does not define media_setting.json file this mechanism would not kick-in.